File: frameworks/domain.core/src/test/java/com/qomolangma/frameworks/domain/core/FixedIdGeneratorTest.java
package com.qomolangma.frameworks.domain.core;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class FixedIdGeneratorTest {
    public static final String FIXED = "fixed";

    @Test
    void should_generate_fixed_identity() {
        Id id = new FixedId(FIXED);
        assertThat(id.next()).isEqualTo(FIXED);
        assertThat(id.nextReadable()).isEqualTo(FIXED);
    }
}

File: frameworks/domain.core/src/test/kotlin/com/qomolangma/frameworks/domain/core/FixedIdGeneratorTest.kt
package com.qomolangma.frameworks.domain.core

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class FixedIdGeneratorTest {
    @Test
    fun should_generate_fixed_identity() {
        val idGenerator: IdGenerator = FixedIdGenerator(FIXED)
        assertThat(idGenerator.next()).isEqualTo(FIXED)
        assertThat(idGenerator.nextReadable()).isEqualTo(FIXED)
    }

    companion object {
        const val FIXED = "fixed"
    }
}
